includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -853,106 +1069,245 @@ public function wc_setup_payments_save() {
+						<p>
+							<?php esc_html_e( 'Your theme is not compatible with WooCommerce. We recommend you switch over to Storefront, a free WordPress theme built and maintained by the makers of WooCommerce. If enabled, Storefront will be installed and activated for you.', 'woocommerce' ); ?>
+						</p>
+					</div>
+
+					<div class="wc-wizard-service-enable">
+						<span class="wc-wizard-service-toggle">
+							<input id="setup_storefront_theme" type="checkbox" name="setup_storefront_theme" value="yes" checked="checked" />
+							<label for="setup_storefront_theme">
+						</span>
+					</div>
+				</li>
 			</ul>
+			<?php endif; ?>
+			<?php if ( $this->should_show_automated_tax_extra() ) : ?>
+				<ul class="wc-wizard-services featured">
+					<li class="wc-wizard-service-item">
+						<div class="wc-wizard-service-description">
+							<h3><?php esc_html_e( 'Automated Taxes', 'woocommerce' ); ?></h3>
+							<p>
+								<?php esc_html_e( 'We’ll automatically calculate and charge the correct rate of tax for each time a customer checks out.', 'woocommerce' ); ?>
+							</p>
+						</div>
+
+						<div class="wc-wizard-service-enable">
+						<span class="wc-wizard-service-toggle">
+							<input id="setup_automated_taxes" type="checkbox" name="setup_automated_taxes" value="yes" checked="checked" />
+							<label for="setup_automated_taxes">
+						</span>
+						</div>
+					</li>
+				</ul>
+			<?php endif; ?>
 			<p class="wc-setup-actions step">
-				<input type="submit" class="button-primary button button-large button-next" value="<?php esc_attr_e( 'Install &amp; activate Storefront', 'woocommerce' ); ?>" name="save_step" />
-				<a href="<?php echo esc_url( $this->get_next_step_link() ); ?>" class="button button-large button-next"><?php esc_html_e( 'Skip this step', 'woocommerce' ); ?></a>
+				<input type="submit" class="button-primary button button-large button-next" value="<?php esc_attr_e( 'Continue', 'woocommerce' ); ?>" name="save_step" />
 				<?php wp_nonce_field( 'wc-setup' ); ?>
 			</p>
 		</form>
 		<?php
-
 	}
 
 	/**
-	 * Theme step save.
+	 * Extras step save.
 	 */
-	private function wc_setup_theme_save() {
+	public function wc_setup_extras_save() {
 		check_admin_referer( 'wc-setup' );
-		wp_schedule_single_event( time() + 1, 'woocommerce_theme_background_installer', array( 'storefront' ) );
+
+		$setup_automated_tax = isset( $_POST['setup_automated_taxes'] ) && 'yes' === $_POST['setup_automated_taxes'];
+		$install_storefront  = isset( $_POST['setup_storefront_theme'] ) && 'yes' === $_POST['setup_storefront_theme'];
+
+		if ( $setup_automated_tax ) {
+			update_option( 'woocommerce_calc_taxes', 'yes' );
+
+			$this->install_woocommerce_services();
+
+			// Signal WooCommerce Services to setup automated taxes.
+			update_option( 'woocommerce_setup_automated_taxes', true );
+		}
+
+		if ( $install_storefront ) {
+			wp_schedule_single_event( time() + 1, 'woocommerce_theme_background_installer', array( 'storefront' ) );
```

Comment:
Using the "fake WordPress cron" for this seems hacky at best, and quite fragile at worst. It depends on a page being loaded after it's scheduled, and according to [this](https://wordpress.org/support/topic/scheduled-posts-still-not-working-in-282/#post-1175405) it's dependent on the server being properly configured.

I suggest doing this instead:
```
// flush any buffers and send the headers
while ( @ob_end_flush() );
flush();
// Prevent timeouts
ignore_user_abort(true);
// Silence any further print to stdout/stderr (I don't remember the function, ob_something())
// Install the plugin / theme syncronously
```

Same for the rest of the plugin / theme installs.